docs/API.md

```diff
@@ -80,11 +80,66 @@ export default Route.extend({
   model() {
     return this.store.findRecord('user', this.get('session.data.authenticated.userId'));
   },
-  
+
   afterModel(model) {
     if (model.get('organizations.length') > 0) {
       return this.transitionTo('board');
     }
   }
 });
-```
\ No newline at end of file
+```
+
+## Feature Toggles
+
+### 🗺️ Problème
+
+On veut pouvoir mettre en prod une fonctionnalité (exemple "Certification v2"),
+tout en ayant la capacité de la désactiver sans générer une nouvelle version de
+l'application pendant les premières semaines.
+
+Sur des fonctionnalités longues à développer, on a aussi parfois le besoin de
+livrer une partie du code en production sans que la fonctionnalité associée
+soit visible par l'utilisateur.
+
+### 🥚 Solution
+
+Ajouter des variables d'environnement dont le nom est clairement identifié
+comme _feature toggle_ en utilisant le préfixe `FT_` :
+
+- `FT_ACTIVATE_CERTIFICATION_V2 = true`
+
+Dans leur formulation, les variables d'env sont en tournure affirmative :
+
+- `FT_USE_ONLY_V1_CERTIFICATION` plutôt que `FT_DONT_USE_V2_CERTIFICATION`.
+
+Leur valeur par défaut est `false` de préférence (=> oublier de l'ajouter
+conserve le comportement actuel de l'application).
+
+⚠️ : il est important de supprimer ces bascules au plus tôt, dès que la feature
+est bien installée en production (voir référence de Martin Fowler ci-dessous).
+Les _features toggles_ ne sont pas des configurations qu'on souhaite conserver
+longtemps, ce sont des bascules temporaires.
+
+Avoir un préfixe bien identifié permet de faire la différence entre les
+variables d'environnement de configuration durables et les variables
+d'environnement de bascules temporaires.
+
+⚠️ : cette solution de _feature toggle_ doit rester un pis aller quand on ne
+sait pas découper finement une fonctionnalité. Ça ne doit pas devenir un
+réflexe. On ne le fait que quand on n'a pas trouvé de meilleure solution.
+
+### 📖 Informations supplémentaires
+
+Pour tester en évitant une combinatoire ingérable pendant les tests (telle
+fonctionnallité activée avec telle autre désactivée, ...), Martin Fowler
+propose de ne tester que deux cas :
+
+1. Tester avec toutes les bascules qui seront effectivement activées lors de la
+   prochaine livraison
+1. Tester avec toutes les bascules activées
```

Comment:
J'ai du mal à voir la différence exacte entre les deux propositions.

Autres : 
- Le 2. est noté 1.
- Les tournures de phrases comme "Martin Fowler propose..." me fait poser deux questions : est-ce que l'équipe est OK avec la proposition de martin. Est-ce que martin fait partie de l'équipe ? (si on ne connait pas le monsieur)

Reply:
Les bascules servent à déployer une fonctionnalité sans l'activer, donc en prod il y aura des bascules désactivées, et d'autres activités. Alors on se pose parfois la question de quoi tester, chaque bascule activée isolément, toutes les combinaisons, ... ? En fait, en général tester les combinaisons citées ci-dessus suffit.

Reply:
Pour le 1. suivi de 1., je fais comme ça en markdown pour une liste numérotée.

Reply:
Oui, on n'est pas obligé de suivre Martin, vous en pensez quoi ?

Reply:
Sinon j'aime bien Martin, demandez-moi si vous ne connaissez pas.

Reply:
Je ne le connais pas, mais je suppose que c'est quelqu'un de bien :)
Ma remarque c'est que si des devs ne sont pas au courant de son existence, ma première réaction à la lecture c'est que Martin est un ancien dev pix :) 

Sinon, je comprends le pourquoi des tests de bascule` c'est juste qu'il me manque possiblement un exemple de ce que l'on test.
Si on ajoute `FT_ACTIVATE_CERTIFICATION_V2`, alors on test avec `FT_ACTIVATE_CERTIFICATION_V2` a `true` seulement, etc...

Reply:
Merci, ajouté !

Et je recommande fortement toute la production de Martin Fowler, j'ai appris énormément grâce à ses articles, livres, et vidéos, et je connais beaucoup de gens pour qui c'est également le cas.